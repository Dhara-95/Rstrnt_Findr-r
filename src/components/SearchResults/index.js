import React from "react";
import { searchResultQueryLabel } from "../../Labels";

const SearchResults = (props) => {
  const { userInput } = props;
  return (
    <div>
      <h2 className="text-grey">
        {searchResultQueryLabel}
        {userInput}
      </h2>
    </div>
  );
};

export default SearchResults;
