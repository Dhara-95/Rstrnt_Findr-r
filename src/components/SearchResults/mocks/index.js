import { RESTAURANT_BY_POSTCODE } from "../../../queries";

const mocks = [
  {
    request: {
      query: RESTAURANT_BY_POSTCODE,
      variables: { postcode: "W13 34H" },
    },
    result: {
      data: {
        dog: {
          name: "Douglas",
        },
      },
    },
  },
];

export { mocks };
