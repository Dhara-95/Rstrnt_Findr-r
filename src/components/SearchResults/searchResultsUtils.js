import { RESTAURANT_BY_POSTCODE } from "../../queries";

const fetchRestaurantData = async (userInput, updateRestaurantData) => {
  try {
    const rawResponse = await fetch(
      "https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/graphql",
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
          Authorization: process.env.REACT_APP_YELP_API_KEY,
          "Accept-Language": "en-US",
        },
        body: JSON.stringify({
          query: RESTAURANT_BY_POSTCODE,
          variables: {
            postcode: userInput,
          },
        }),
      }
    );
    const content = await rawResponse.json();
    const { business } = content.data.search;
    updateRestaurantData(business);
  } catch (error) {
    console.log("There was an error with the fetch: " + error);
  }
};

export { fetchRestaurantData };
