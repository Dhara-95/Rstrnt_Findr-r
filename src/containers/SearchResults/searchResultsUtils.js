import { LabelsObject } from "../../Labels";

import { RESTAURANT_BY_POSTCODE } from "../../queries";

const fetchRestaurantData = async (
  userInput,
  updateRestaurantData,
  updateLoadingState
) => {
  updateLoadingState(true);
  try {
    const rawResponse = await fetch(
      "https://rstrnt-cors.herokuapp.com/https://api.yelp.com/v3/graphql",

      {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
          Authorization: process.env.REACT_APP_YELP_API_KEY,
          "Accept-Language": "en-US",
        },
        body: JSON.stringify({
          query: RESTAURANT_BY_POSTCODE,
          variables: {
            postcode: userInput,
          },
        }),
      }
    );
    const content = await rawResponse.json();
    const { business } = content.data.search;
    updateRestaurantData(business);
  } catch (error) {
    console.log("There was an error with the fetch: " + error);
  }
  updateLoadingState(false);
};

const checkValidImageURL = (url) => {
  const { invalidImageURLLabel } = LabelsObject;

  if (url === "https://s3-media3.fl.yelpcdn.com/bphoto/None/o.jpg") {
    return invalidImageURLLabel;
  } else {
    return url;
  }
};

export { fetchRestaurantData, checkValidImageURL };
