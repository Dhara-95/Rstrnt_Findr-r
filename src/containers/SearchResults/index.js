import React, { useEffect, useState } from "react";

import { filterByRating } from "../../utils/filterByRating";
import { LoadingConfirmation, RestaurantsNear } from "../../components";
import { RestaurantList } from "../index";
import { yelp } from "../../utils/fetchRestaurantData";

const SearchResults = (props) => {
  const { userInput } = props;
  const [restaurantData, updateRestaurantData] = useState([]);
  const [isLoading, updateLoadingState] = useState(false);
  const [defaultRestaurantOrder, applyDefaultRestaurantOrder] = useState([]);

  useEffect(() => {
    yelp.getData(userInput, updateRestaurantData, updateLoadingState);
    // updateLoadingState only works within the method call - need to look into
  }, [userInput]);

  const handleOnChange = (action) => {
    if (!defaultRestaurantOrder.length) {
      applyDefaultRestaurantOrder([...restaurantData]);
    }
    const actionList = {
      rating: filterByRating(restaurantData),
      nearest: [...defaultRestaurantOrder],
    };

    updateRestaurantData(actionList[action]);
  };

  return (
    <>
      <RestaurantsNear userInput={userInput} />
      {isLoading && <LoadingConfirmation />}
      <RestaurantList data={restaurantData} handleOnChange={handleOnChange} />
    </>
  );
};

export { SearchResults };
