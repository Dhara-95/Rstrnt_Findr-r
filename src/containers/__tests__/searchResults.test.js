import React from "react";
import renderer from "react-test-renderer";

import { SearchResults } from "../SearchResults";

it("displays the user input correctly", async () => {
  const tree = renderer
    .create(<SearchResults userInput={"W13 3RH"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
