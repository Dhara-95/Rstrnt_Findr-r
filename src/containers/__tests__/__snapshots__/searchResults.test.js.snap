// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the user input correctly 1`] = `
Array [
  <h2
    className="text-grey"
  >
    Restaurants near 
    W13 3RH
  </h2>,
  <div
    className="flex flex-col"
  >
    <div
      className=" list-options flex flex-col items-center lg:justify-center lg:flex-row xl:justify-center xl:flex-row 2xl:justify-center 2xl:flex-row p-4  w-1/2 mx-auto"
    >
      <h2
        className="text-grey m-4"
      >
        Order by:
      </h2>
      <select
        className="appearance-none rounded-full w-full bg-white lg:w-1/2 xl:w-1/2 2xl:w-1/3 h-8"
        id="filter"
        name="filter"
        onChange={[Function]}
      >
        <option
          value="nearest"
        >
          Nearest to you
        </option>
        <option
          value="rating"
        >
          By rating
        </option>
      </select>
    </div>
  </div>,
]
`;
